compiler_opt/tools/generate_default_trace.py

```diff
@@ -160,74 +145,60 @@ def main(_):
   # other smaller files are processed in parallel
   corpus_elements = cps.sample(k=sampled_modules, sort=True)
 
-  worker_count = (
-      min(os.cpu_count(), _NUM_WORKERS.value)
-      if _NUM_WORKERS.value else os.cpu_count())
-
   tfrecord_context = (
       tf.io.TFRecordWriter(_OUTPUT_PATH.value)
       if _OUTPUT_PATH.value else contextlib.nullcontext())
   performance_context = (
       tf.io.gfile.GFile(_OUTPUT_PERFORMANCE_PATH.value, 'w')
       if _OUTPUT_PERFORMANCE_PATH.value else contextlib.nullcontext())
+  work = [
+      cps.load_module_spec(corpus_element) for corpus_element in corpus_elements
+  ]
 
   with tfrecord_context as tfrecord_writer:
     with performance_context as performance_writer:
-      ctx = multiprocessing.get_context()
-      m = ctx.Manager()
-      results_queue: 'queue.Queue[ResultsQueueEntry]' = m.Queue()
-      work_queue: 'queue.Queue[corpus.LoadedModuleSpec]' = m.Queue()
-      for corpus_element in corpus_elements:
-        work_queue.put(cps.load_module_spec(corpus_element))
-
-      # pylint:disable=g-complex-comprehension
-      processes = [
-          ctx.Process(
-              target=functools.partial(worker, _POLICY_PATH.value, work_queue,
-                                       results_queue, _KEY_FILTER.value))
-          for _ in range(0, worker_count)
-      ]
-      # pylint:enable=g-complex-comprehension
-
-      for p in processes:
-        p.start()
-
-      total_successful_examples = 0
-      total_work = len(corpus_elements)
-      total_failed_examples = 0
-      total_training_examples = 0
-      for _ in range(total_work):
-        logging.log_every_n_seconds(logging.INFO,
-                                    '%d success, %d failed out of %d', 10,
-                                    total_successful_examples,
-                                    total_failed_examples, total_work)
-
-        results = results_queue.get()
-        if isinstance(results, BaseException):
-          logging.fatal(results)
-        if not results:
-          total_failed_examples += 1
-          continue
-
-        total_successful_examples += 1
-        module_name, records, reward_stat = results
-        if tfrecord_writer:
-          total_training_examples += len(records)
-          for r in records:
-            tfrecord_writer.write(r)
-        if performance_writer:
-          for key, value in reward_stat.items():
-            performance_writer.write(
-                (f'{module_name},{key},{value.default_reward},'
-                 f'{value.moving_average_reward}\n'))
-
-      print((f'{total_successful_examples} of {len(corpus_elements)} modules '
-             f'succeeded, and {total_training_examples} trainining examples '
-             'written'))
-      for p in processes:
-        p.join()
+      with local_worker_manager.LocalWorkerPoolManager(
```

Comment:
can you abstract this away (similar to train_locally.py) so we can use larger worker pools for generating traces?

Reply:
done